File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br= new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st= new StringTokenizer(br.readLine());
        StringBuilder sb= new StringBuilder("");

        int n=Integer.parseInt(st.nextToken());
        int k=Integer.parseInt(st.nextToken());
        int cnt=0;

        ArrayList<Integer> answer=new ArrayList<>();
        Queue<Integer> ycps=new LinkedList<>();

        for (int i = 1; i < n+1; i++) {
            ycps.offer(i);
        }

        while (!ycps.isEmpty()){
            int q=ycps.poll();
            cnt++;
            if(cnt==k){
                answer.add(q);
                cnt=0;
                continue;
            }
            ycps.offer(q);
        }

        sb.append("<");
        for (int i = 0; i < n; i++) {
            if (i==n-1){
                sb.append(answer.get(i)+">");
                continue;
            }
            sb.append(answer.get(i)+", ");
        }

        System.out.println(sb);
    }
}

File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.py
from collections import deque

n,k=map(int,input().split())
ycps=deque(i for i in range(1,n+1))
answer=[]
cnt=0

while ycps:
    q=ycps.popleft()
    cnt+=1
  
    if cnt==k:
        #제거
        answer.append(q)
        cnt=0
        continue
    ycps.append(q)

print("<",end="")
for i in range(len(answer)):
    if i==len(answer)-1: #마지막인덱스
        print("%d"%answer[i],end="")
        continue
    print("%d, " %answer[i] ,end="")
print(">")